### PR TITLE
Handles and/or documents pres client errors.

### LINKED ISSUE #545: trap for preservation-client gem errors in dor-services-app and handle accordingly
dor-services-app uses preservation-client in the following ways:
- shelving API service:
   - uses preservation-client to get content diff for shelved files
- versioning service:
   - uses preservation-client to get current_version
- other?

dor-services-app code should trap for any errors from preservation-client (e.g. can't connect, not found, generic error, response status other than expected (i.e. not a 200) ) and:
- pass along appropriate error code to dor-services-client (which needs to handle it properly)
- notify honeybadger as appropriate

See https://github.com/sul-dlss/preservation-client/blob/master/lib/preservation/client.rb#L30-L39

### PR DESCRIPTION
**Pull request by justinlittman**
closes #545

## Why was this change made?
To make sure that preservation client errors are handled correctly.

## Was the API documentation (openapi.json) updated?
No.

### CODE REVIEW

File: app/jobs/create_virtual_objects_job.rb [hunk 1]
```diff
@@ -17,7 +17,7 @@ def perform(virtual_objects:, background_job_result:)
       # Do not add `nil`s to the errors array as they signify successful
       # creation of the virtual object
       errors << result if result.present?
-    rescue ActiveFedora::ObjectNotFoundError, Rubydora::FedoraInvalidRequest, Dor::Exception => e
+    rescue ActiveFedora::ObjectNotFoundError, Rubydora::FedoraInvalidRequest, Dor::Exception, Preservation::Client::Error => e
```
Comment (mjgiarlo) [inline app/jobs/create_virtual_objects_job.rb]: I wonder what folks would think about dropping lines 20 and 21 altogether? I'm not sure that they add anything, as all four of these exception classes are subclasses of `StandardError`, so lines 22 and 23 would handle them the exact same way without altering any behavior. See:

```ruby
[1] pry(main)> ActiveFedora::ObjectNotFoundError < StandardError
=> true
[2] pry(main)> Rubydora::FedoraInvalidRequest < StandardError
=> true
[3] pry(main)> Dor::Exception < StandardError
=> true
[4] pry(main)> Preservation::Client::Error < StandardError
=> true
```

File: app/services/constituent_service.rb [hunk 2]
```diff
@@ -19,6 +19,7 @@ def initialize(parent_druid:)
   # subsequent calls will erase the previous changes.
   # @param [Array<String>] child_druids the identifiers of the child objects
   # @raises ActiveFedora::RecordInvalid if AF object validations fail on #save!
+  # @raise [Preservation::Client::Error] if bad response from preservation catalog.
```
Comment (mjgiarlo) [inline app/services/constituent_service.rb]: `s/@raise/@raises/` for consistency w/ prior line (and wouldn't mind seeing the exception in the prior line wrapped in brackets!)

Comment (mjgiarlo) [reply app/services/constituent_service.rb]: On second thought, it looks like there's more usage of `@raises` (14 instances) than `@raise` (8 instances) in this codebase. Maybe best to standardize on `@raises`, assuming both are OK with YARD? Meh.

File: app/jobs/create_virtual_objects_job.rb (hunk 1 above)
Comment (jcoyne) [reply app/jobs/create_virtual_objects_job.rb]: I prefer we keep this explicit, so we have some idea about exactly what we're expecting.

File: app/services/constituent_service.rb (hunk 2 above)
Comment (jcoyne) [reply app/services/constituent_service.rb]: I think `@raise` is corrrect.

Comment (mjgiarlo) [reply app/services/constituent_service.rb]: Good for a follow-on PR then, to be both consistent and correct? :laughing:

**Pull request merged**